contents/docs/data/identify.mdx

```diff
@@ -0,0 +1,466 @@
+
+> We don't recommend using emails as distinct IDs, as people changing their email will lead to extra complexity to maintaining data integrity. It is better to use database IDs or account IDs.
+
+On your frontend, you should call identify as soon as you're able to. Typically, this is every time your app loads for the first time, or directly after your users log in. This ensures that events sent during your users' sessions are correctly associated with them. 
+
+### Setting user properties
+
+You'll notice that one of the parameters in the `identify` method is a `properties` object. This enables you to set [user properties](/docs/getting-started/user-properties), which can be used in PostHog features such as [filters](/docs/product-analytics/trends#filtering-events-based-on-properties) or [cohorts](/docs/data/cohorts). 
+
+> Whenever possible, we recommend passing in all user properties you have available each time you call identify, as this will ensure that their user profile on PostHog is up to date.
+
+User properties can also be set using a `capture` and not only with `identify`. See our [user properties docs](/docs/getting-started/user-properties) for more details on how to work with them and best practices.
+
+## Alias
+
+Sometimes, you may want to assign multiple distinct IDs to a single user. This is helpful in scenarios where your primary distinct ID may be inaccessible. For example, if a distinct ID which is typically used on the frontend is not available in certain parts of your backend code. In this case, you can use `alias` to assign another distinct ID to the same user:
+
+In the below example, we assign the user with `distinct_id` another ID: `alias_id`. This means that any events submitted using either `distinct_id` or `alias_id` will be associated with the same user.
+
+<MultiLanguage selector="tabs">
+
+```node
+client.alias({
+    distinctId: 'distinct_id',
+    alias: 'alias_id',
+})
+```
+
+```javascript
+posthog.alias('alias_id', 'distinct_id');
+```
+
+```python
+posthog.alias('alias_id', 'distinct_id');
+```
+
+```php
+PostHog::alias(array(
+  'distinctId' => 'distinct_id',
+  'alias' => 'alias_id'
+));
+```
+
+```ruby
+posthog.alias({
+  distinct_id: "distinct_id",
+  alias: "alias_id",
+})
+```
+
+```go
+client.Enqueue(posthog.Alias{
+  DistinctId: "distinct_id",
+  Alias: "alias_id",
+})
+```
+
+```java
+posthog.alias("distinct_id", "alias_id");
+```
+
+```bash
+curl -v -L --header "Content-Type: application/json" -d '{
+    "api_key": "<ph_project_api_key>",
+    "properties": {
+        "distinct_id": "distinct_id_one",
+        "alias": "any_other_id"
+    },
+    "timestamp": "2020-08-16 09:03:11.913767",
+    "event": "$create_alias"
+}' https://app.posthog.com/capture/
+```
+
+</MultiLanguage>
+
+There are two requirements when assigning an `alias_id`:
+
+1. It cannot be associated with more than one `distinct_id`.
+2. The `alias_id` **must not** be the `distinct_id` of any other user. For example:
```

Comment:
What's missing?

Reply:
if you have called `$identify(distinct_id, ...)` anywhere even in the backend and without another id, then you can't use that distinct_id as alias_id anymore. The order in which you provide id's is important. I think the previous example was better we had, it's more clear how someone would actually run into it
```
identify('distinct_id')
alias(distinct_id = 'new_id', alias_id = 'distinct_id')  // doesn't work
alias(distinct_id = 'distinct_id;, alias_id = 'new_id') // does work assuming 'new_id' isn't used anywhere.
```

Reply:
Hmm interesting. I agree the previous example was clearer, but I removed it since we are encouraging users not to call identify in the backend (since it doesnt really make sense to use it in the backend), so this is why I removed `identify` in these examples.

I dont feel too strongly about this though. Happy to change it back if you prefer. What do you think?

Reply:
you could just say imagine in the past we've called this on a separate session in front of the examples.

Reply:
The actual restriction (that I'm about to change the code to have) is if you've used this distinct_id as distinct_id for ('$identify', '$create_alias' or '$merge_dangerously') then you can'd use it as anon nor alias ID anymore

Reply:
Hmm I updated the example to be similar to what it was previously. What do you think? 
I think we should keep it in this section to help paint a picture of how to use alias